spec/controllers/users/mfa_selection_controller_spec.rb

```diff
@@ -0,0 +1,156 @@
+require 'rails_helper'
+
+describe Users::MfaSelectionController do
+  before do
+    allow(IdentityConfig.store).to receive(:select_multiple_mfa_options).and_return(true)
+    user = build(:user, :signed_up)
+    stub_sign_in(user)
+  end
+  let(:current_sp) { create(:service_provider) }
+
+  describe 'GET index' do
```

Comment:
each of the `#update` specs seems to stub sign in on their own, so we don't need this at the top of the file, we only need it for `#index` now

```suggestion
  end
  let(:current_sp) { create(:service_provider) }

  describe 'GET index' do
    before do
      user = build(:user, :signed_up)
      stub_sign_in(user)
    end
```